src/data_warehouse/data/edge.ts

```diff
@@ -0,0 +1,153 @@
+import {BaseDomainClass} from "../../common_classes/base_domain_class";
+import {IsBoolean, IsNotEmpty, IsNumber, IsObject, IsOptional, IsString, IsUUID, ValidateIf} from "class-validator";
+import {Expose, plainToClass, Transform} from "class-transformer";
+import Container from "../ontology/container";
+import MetatypeRelationshipPair, {MetatypeRelationshipPairID} from "../ontology/metatype_relationship_pair";
+import Node from "./node";
+
+/*
+    Edge represents an edge record in the Deep Lynx database and the various
+    validations required for said record to be considered valid.
+ */
+export default class Edge extends BaseDomainClass {
+    @IsOptional()
+    @IsUUID()
+    id?: string
+
+    @IsUUID()
+    container_id?: string
+
+    @IsOptional()
+    @IsBoolean()
+    archived?: boolean
+
+    // we often need the metatype's name, it's keys, or access to other properties
+    // when we deal with nodes, so for ease of use we're going to use the whole
+    // class, not just the id
+    @MetatypeRelationshipPairID({message: "Metatype relationship pair must have valid ID"})
+    @Expose({name:"relationship_pair_id", toClassOnly: true})
+    @Transform(({value}) => {
+        const p = plainToClass(MetatypeRelationshipPair, {})
+        p.id = value
+        return p
+    }, {toClassOnly: true})
+    metatypeRelationshipPair: MetatypeRelationshipPair | undefined
+
+    // these two getters are to maintain the original api response for nodes
+    @Expose({toPlainOnly: true})
+    get relationship_pair_id(): string {
+        return (this.metatypeRelationshipPair) ? this.metatypeRelationshipPair.id! : ""
+    }
+
+    @IsObject()
+    properties: object = {}
+
+    @IsString()
+    @IsOptional()
+    original_data_id?: string
+
+    @IsString()
+    @IsOptional()
+    composite_original_id?: string
+
+    @IsUUID()
+    @IsOptional()
+    import_data_id?: string
+
+    @IsNumber()
+    @IsOptional()
+    data_staging_id?: number
+
+    // if we have an composite id, we must provide a data source as we can only
+    // guarantee uniqueness based on that combination of values
+    @ValidateIf(o =>  typeof o.composite_original_id !== "undefined" && o.composite_original_id !== null)
+    @IsUUID()
+    data_source_id?: string
+
+    @IsUUID()
+    @IsOptional()
+    type_mapping_transformation_id?: string // TODO: fetch the full type mapping object
```

Comment:
Should this TODO be resolved before PR completion?

Reply:
Decided this was not a good idea. Went ahead and removed it.